x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -119,16 +119,14 @@ public void onFailure(Exception e) {
     }
 
     private Map<String, Object> getMappings(final GetIndexResponse getIndexResponse, final String sourceIndexName) {
-        ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetaData>> mappings = getIndexResponse.mappings();
-        ImmutableOpenMap<String, MappingMetaData> indexMapping = mappings.get(sourceIndexName);
-        if (indexMapping.keys().size() == 0) {
+        ImmutableOpenMap<String, MappingMetaData> mappings = getIndexResponse.mappings();
+        MappingMetaData indexMapping = mappings.get(sourceIndexName);
+        if (indexMapping == null || indexMapping == MappingMetaData.EMPTY_MAPPINGS) {
```

Comment:
I'm wondering why we need to check for `null` here?

Reply:
Because the index itself might not exist for some reason, in which case `Map.get(key)` will return `null`

Reply:
Although looking more closely, the GetIndexRequest used above here will error out if the index doesn't exist, so we can remove the null check.  Good catch!